internal/core/data/device_test.go

```diff
@@ -344,7 +345,7 @@ func buildReadings() []models.Reading {
 		Pushed:   ticks,
 		Device:   testDeviceName}
 
-	r2 := models.Reading{Id: bson.NewObjectId(),
+	r2 := models.Reading{Id: bson.NewObjectId().Hex(),
```

Comment:
I left this one as-is because we are designing the API to accept both but only generate one. I'll leave a comment in the code since that's fairly non-obvious.